libraries/core_libs/consensus/src/pbft/pbft_manager.cpp

```diff
@@ -718,20 +718,22 @@ bool PbftManager::updateSoftVotedBlockForThisRound_() {
   if (!soft_voted_block_for_this_round_.second) {
     auto round = getPbftRound();
 
-    auto voted_block_hash_with_soft_votes = vote_mgr_->getVotesBundleByRoundAndStep(round, 2, TWO_T_PLUS_ONE);
-
     auto batch = db_->createWriteBatch();
-    db_->addPbftMgrVotedValueToBatch(PbftMgrVotedValue::SoftVotedBlockHashInRound,
-                                     voted_block_hash_with_soft_votes.voted_block_hash, batch);
-    db_->addPbftMgrStatusToBatch(PbftMgrStatus::SoftVotedBlockInRound, voted_block_hash_with_soft_votes.enough, batch);
-    if (voted_block_hash_with_soft_votes.enough &&
-        voted_block_hash_with_soft_votes.voted_block_hash != NULL_BLOCK_HASH) {
-      db_->addSoftVotesToBatch(round, voted_block_hash_with_soft_votes.votes, batch);
+    if (auto voted_block_hash_with_soft_votes = vote_mgr_->getVotesBundleByRoundAndStep(round, 2, TWO_T_PLUS_ONE)) {
+      db_->addPbftMgrVotedValueToBatch(PbftMgrVotedValue::SoftVotedBlockHashInRound,
+                                       voted_block_hash_with_soft_votes->voted_block_hash, batch);
+      db_->addPbftMgrStatusToBatch(PbftMgrStatus::SoftVotedBlockInRound, true, batch);
+      if (voted_block_hash_with_soft_votes->voted_block_hash != NULL_BLOCK_HASH) {
+        db_->addSoftVotesToBatch(round, voted_block_hash_with_soft_votes->votes, batch);
+      }
+      soft_voted_block_for_this_round_ = {voted_block_hash_with_soft_votes->voted_block_hash, true};
+    } else {
+      db_->addPbftMgrVotedValueToBatch(PbftMgrVotedValue::SoftVotedBlockHashInRound, NULL_BLOCK_HASH, batch);
+      db_->addPbftMgrStatusToBatch(PbftMgrStatus::SoftVotedBlockInRound, false, batch);
+      soft_voted_block_for_this_round_ = {NULL_BLOCK_HASH, false};
```

Comment:
I think that should be:
if (auto voted_block_hash_with_soft_votes = vote_mgr_->getVotesBundleByRoundAndStep(round, 2, TWO_T_PLUS_ONE)) {
      auto batch = db_->createWriteBatch();
      db_->addPbftMgrVotedValueToBatch(PbftMgrVotedValue::SoftVotedBlockHashInRound,
                                       voted_block_hash_with_soft_votes->voted_block_hash, batch);
      db_->addPbftMgrStatusToBatch(PbftMgrStatus::SoftVotedBlockInRound, true, batch);
      db_->addSoftVotesToBatch(round, voted_block_hash_with_soft_votes->votes, batch);
      db_->commitWriteBatch(batch);
      soft_voted_block_for_this_round_ = {voted_block_hash_with_soft_votes->voted_block_hash, true};
}

Reply:
but before we did changes based on return value of `getVotesBundleByRoundAndStep`. So I do no need to set anything in case we do not have enough votes to specify any round?

Reply:
When PBFT go to a new round(in resetRound_), that will reset soft voted value to NULL. 
https://github.com/Taraxa-project/taraxa-node/blob/develop/libraries/core_libs/consensus/src/pbft/pbft_manager.cpp#L399
https://github.com/Taraxa-project/taraxa-node/blob/develop/libraries/core_libs/consensus/src/pbft/pbft_manager.cpp#L417

 getVotesBundleByRoundAndStep return values or null pointer. If no value return, don't need set NULL